base/abstractarray.jl

```diff
@@ -1584,17 +1584,18 @@ end
 _cat(dims, X...) = cat_t(promote_eltypeof(X...), X...; dims=dims)
 
 @inline cat_t(::Type{T}, X...; dims) where {T} = _cat_t(dims, T, X...)
-@inline function _cat_t(dims, T::Type, X...)
+@inline function _cat_t(dims, ::Type{T}, X...) where {T}
     catdims = dims2cat(dims)
-    shape = cat_shape(catdims, map(cat_size, X))
+    shape = cat_shape(catdims, map(cat_size, X)::Tuple{Vararg{Union{Int,Dims}}})::Dims
```

Comment:
Looking at `cat_size` it seems to me that it would be just `::Union{Int,Dims}`, but maybe I'm missing something.

Reply:
It may not be flattened (yet).

Reply:
Oh, I get your point. Yes, that's essentially what this type-assert says: each `cat_size(X[i])::Union{Int,Dims}`.

Reply:
Got it, I somehow missed on the `map` part. LGTM